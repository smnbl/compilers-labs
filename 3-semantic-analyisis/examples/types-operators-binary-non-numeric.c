int only_for_numeric_types()
{
    "cannot" + "add strings";
}
