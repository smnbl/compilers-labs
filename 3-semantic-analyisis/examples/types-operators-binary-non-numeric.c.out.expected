sema: error: 3:14: Invalid operand types to binary operator '+'
└── Program <6>
    └── FuncDecl: returnType = 'int', name = 'only_for_numeric_types' <5>
        └── CompoundStmt <4>
            └── ExprStmt <3>
                └── BinaryOpExpr: op = '+' <2>
                    ├── StringLiteral: value = 'cannot' <0>
                    └── StringLiteral: value = 'add strings' <1>
